test/simple_select.test.js

```diff
@@ -0,0 +1,348 @@
+
+  var afterNextRender = AfterNextRender(map);
+
+  var cleanUp = function(cb) {
+    Draw.deleteAll();
+    map.fire.reset();
+    afterNextRender(cb);
+  }
+
+  var getFireArgs = function() {
+    var args = [];
+    for (var i=0; i<map.fire.callCount; i++) {
+      args.push(map.fire.getCall(i).args);
+    }
+    return args;
+  }
+
+  t.test('simple_select - setup', t => {
+    var donedone = false;
+    var done = function() {
+      if (donedone === false) {
+        t.pass('map loaded');
+        t.end();
+        donedone = true;
+      }
+    };
+    if (map.loaded()) {
+      done();
+    }
+    map.on('load', done);
+  });
+
+
+  t.test('simple_select box select', t => {
+    Draw.add(cloneFeature('negitivePoint'));
+    var id = Draw.add(cloneFeature('point'));
+    map.fire.reset();
+
+    afterNextRender(() => {
+      map.fire('mousedown', makeMouseEvent(0, 0, true));
+      map.fire('mousemove', makeMouseEvent(7, 7, true));
+      map.fire('mousemove', makeMouseEvent(15, 15, true));
+      map.fire('mouseup', makeMouseEvent(15, 15, true));
+
+      afterNextRender(() => {
+        var args = getFireArgs().filter(arg => arg[0] === 'draw.select');
+        t.equal(args.length, 1, 'should have one and only one selected start event');
+        if (args.length > 0) {
+          t.equal(args[0][1].featureIds.length, 1, 'should select only one feautre');
+          t.equal(args[0][1].featureIds[0], id, 'should select the feature we expect it to select');
+        }
+        cleanUp(() => t.end());
+      });
+    });
+  });
+
+  t.test('simple_select - box select over 1001 points', t => {
+
+    for(var i=0; i<1001; i++) {
+      Draw.add(cloneFeature('point'));
+    }
+    map.fire.reset();
+
+    afterNextRender(() => {
+      map.fire('mousedown', makeMouseEvent(0, 0, true));
+      map.fire('mousemove', makeMouseEvent(7, 7, true));
+      map.fire('mousemove', makeMouseEvent(15, 15, true));
+      map.fire('mouseup', makeMouseEvent(15, 15, true));
+
+      afterNextRender(() => {
+        t.equal(getFireArgs().filter(arg => arg[0] === 'draw.select').length, 0, 'there should be no draw.select event');
+        cleanUp(() => t.end());
+      });
+    });
+  });
+
+  t.test('simple_select - box select over no points', t => {
+
+    Draw.add(cloneFeature('point'));
+    map.fire.reset();
+
+    afterNextRender(() => {
+      map.fire('mousedown', makeMouseEvent(0, 0, true));
+      map.fire('mousemove', makeMouseEvent(-7, -7, true));
+      map.fire('mousemove', makeMouseEvent(-15, -15, true));
+      map.fire('mouseup', makeMouseEvent(-15, -15, true));
+
+      afterNextRender(() => {
+        t.equal(getFireArgs().filter(arg => arg[0] === 'draw.select').length, 0, 'there should be no draw.select event');
+        cleanUp(() => t.end());
+      });
+    });
+  });
+
+  t.test('simple_select - unselect', t => {
+    var id = Draw.add(cloneFeature('point'));
+    Draw.changeMode('simple_select', [id]);
+    map.fire.reset();
```

Comment:
All these `map.fire.resets()` at the beginning of the case are unnecessary, right? Because of `cleanup()`?